mlir/include/mlir/Dialect/Arith/IR/ArithOps.td

```diff
@@ -1227,17 +1227,32 @@ def Arith_TruncIOp : Arith_IToICastOp<"trunci"> {
 // TruncFOp
 //===----------------------------------------------------------------------===//
 
-def Arith_TruncFOp : Arith_FToFCastOp<"truncf"> {
+def Arith_TruncFOp :
+    Arith_Op<"truncf",
+      [Pure, SameOperandsAndResultShape,
+       DeclareOpInterfaceMethods<ArithRoundingModeInterface>,
+       DeclareOpInterfaceMethods<CastOpInterface>]>,
+    Arguments<(ins FloatLike:$in,
+                   OptionalAttr<Arith_RoundingModeAttr>:$roundingmode)>,
+    Results<(outs FloatLike:$out)> {
   let summary = "cast from floating-point to narrower floating-point";
   let description = [{
     Truncate a floating-point value to a smaller floating-point-typed value.
     The destination type must be strictly narrower than the source type.
-    If the value cannot be exactly represented, it is rounded using the default
-    rounding mode. When operating on vectors, casts elementwise.
+    If the value cannot be exactly represented, it is rounded using the
+    provided rounding mode or the default one if no rounding mode is provided.
+    When operating on vectors, casts elementwise.
   }];
+  let builders = [
+    OpBuilder<(ins "Type":$out, "Value":$in), [{
+      $_state.addOperands(in);
+      $_state.addTypes(out);
+    }]>
+  ];
 
   let hasFolder = 1;
```

Comment:
I think the folder needs to be updated too

Reply:
Makes sense. I'll wait to progress here till I add rounding mode support to the LLVM dialect. That will make code cleaner there. Thanks for pointing out!

Reply:
I can take care of the SPIR-V conversation later on

Reply:
From what I saw, the dialect is missing an enumeration for the rounding modes. The corresponding value for the rounding mode decoration is present already.

Reply:
#86260 is ready for review in case you want to take a look. That's what the `truncf` with rounding mode will lower to.

Reply:
Added conversion to LLVM and fixed folder. Thanks a lot for your comments, @kuhar! 

Are you going to look into conversion to SPIR-V in the end?

Reply:
I don't think we need it immediately for SPIR-V, but I get get it done in a ~month if you file an issue for this.

Reply:
Agree. I created #87050 to track this. Feel free to assign it to yourself. Thanks a lot, @kuhar!